go/nbs/s3_table_persister.go

```diff
@@ -35,63 +40,74 @@ type s3UploadedPart struct {
 	etag string
 }
 
-func (s3p s3TablePersister) Compact(mt *memTable, haver chunkReader) chunkSource {
+func (s3p s3TablePersister) Persist(mt *memTable, haver chunkReader) chunkSource {
 	return s3p.persistTable(mt.write(haver))
 }
 
 func (s3p s3TablePersister) persistTable(name addr, data []byte, chunkCount uint32) chunkSource {
-	if chunkCount > 0 {
-		t1 := time.Now()
-		s3p.multipartUpload(data, name.String())
-		verbose.Log("Compacted table of %d Kb in %s", len(data)/1024, time.Since(t1))
-
-		s3tr := &s3TableReader{s3: s3p.s3, bucket: s3p.bucket, h: name, readRl: s3p.readRl}
-		index := parseTableIndex(data)
-		if s3p.indexCache != nil {
-			s3p.indexCache.put(name, index)
-		}
-		s3tr.tableReader = newTableReader(index, s3tr, s3BlockSize)
-		return s3tr
+	if chunkCount == 0 {
+		return emptyChunkSource{}
 	}
-	return emptyChunkSource{}
+	t1 := time.Now()
+	s3p.multipartUpload(data, name.String())
+	verbose.Log("Compacted table of %d Kb in %s", len(data)/1024, time.Since(t1))
+
+	return s3p.newReaderFromIdxData(data, name)
```

Comment:
nit: "Index". We use "idx" in the code everywhere to mean an index pointer into an array.